### DISCUSSION TITLE
Is Apple M4 Supported?

### DISCUSSION THREAD

**Discussion by RudyD** [tinygrad/tinygrad#7928] [Q&A]
Recently (May 2024) Apple release their M4 chips with M4 Pro and Max later that year (Nov. 2024). Does tinygrad support any of these? Has anyone tried? Before going "all-in" on tinygrad, I'm trying to assess on how much implementation work would be left, if any, to get us there.

**Comment by nimlgen** [comment]
https://docs.tinygrad.org/runtime/

METAL -- M1+ Macs; Metal 3.0+ for bfloat support
